# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/Shmem.h
# Commit: 3d201ea88341
# Full Hash: 3d201ea8834161911890a06bae578b64d18eefa9
# Author: Kelsey Gilbert <jgilbert@mozilla.com>
# Date: 2021-12-15 09:11:58
# Regressor Bug: 1746111
# File Overlap Count: 1
# Description:
#   Bug 1746111 - (Re)serialize WebRTC frame SurfaceDescriptorBuffers for convert-and-upload on WebGLParent. r=gfx-reviewers,nika,lsalzman
#   
#   Also generally de-inline (WebGLParent::TexImage not DispatchCommand)
#   TexImage(SurfaceDescriptor).
#   
# ==============================================================================

diff -r a87211c8eedd -r 3d201ea88341 ipc/glue/Shmem.h
--- a/ipc/glue/Shmem.h	Wed Dec 15 01:49:11 2021 +0000
+++ b/ipc/glue/Shmem.h	Wed Dec 15 01:54:21 2021 +0000
@@ -16,6 +16,7 @@
 #include "nsDebug.h"
 
 #include "mozilla/ipc/SharedMemory.h"
+#include "mozilla/Range.h"
 #include "mozilla/UniquePtr.h"
 
 /**
@@ -132,6 +133,11 @@
     return mSize / sizeof(T);
   }
 
+  template <typename T>
+  Range<T> Range() const {
+    return {get<T>(), Size<T>()};
+  }
+
   // These shouldn't be used directly, use the IPDL interface instead.
   id_t Id(PrivateIPDLCaller) const { return mId; }
 
